nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/AbstractBulkAction.java

```diff
@@ -137,22 +132,31 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
                 loadCurrentBulkCommandContext(commandId);
             }
             // process record
+            boolean startTimer = documentIds.isEmpty();
             documentIds.addAll(docIdsFrom(record));
             if (documentIds.size() >= size) {
                 processBatch(context);
+            } else if (startTimer && !documentIds.isEmpty()) {
+                context.setTimer("timer", System.currentTimeMillis() + timer);
             }
         }
 
         @Override
         public void processTimer(ComputationContext context, String key, long timestamp) {
             processBatch(context);
-            context.setTimer("timer", System.currentTimeMillis() + timer);
         }
 
         @Override
         public void destroy() {
-            getLog().debug(
-                    String.format("Destroy computation: %s, pending entries: %d", metadata.name(), documentIds.size()));
+            getLog().debug(String.format("Destroy: %s, pending entries: %d", metadata.name(), documentIds.size()));
+        }
+
+        /**
+         * Define if the computation produces count records.<br>
+         * Intermediate computations don't want to do it.
+         */
+        protected boolean doesCount() {
```

Comment:
I asked for this because I have intermediate computations.  At the moment, I don't need "another abstract computation".  I already have an implementation based on `AbstractBatchComputation`.  I am just blocked waiting for this.

Reply:
I don't prefer to wait! ;)